### PR TITLE
Put card space preview in the workflow

### PR DESCRIPTION
**Pull request by Aierie**
blank preview state in the create card space workflow

partially filled-in preview state in the create card space workflow

### CODE REVIEW

File: packages/web-client/app/components/card-space/create-space-workflow/index.ts [hunk 1]
```diff
@@ -234,6 +234,18 @@ export default class CreateSpaceWorkflowComponent extends RestorableWorkflowComp
...
 
+  get currentCardSpaceDetails() {
+    return {
+      profilePhoto: this.workflow.session.getValue('profileImageUrl'),
+      coverPhoto: this.workflow.session.getValue('coverPhotoUrl'),
+      name: this.workflow.session.getValue('displayName'),
+      host: this.workflow.session.getValue('url'),
+      category: this.workflow.session.getValue('profileCategory'),
+      description: this.workflow.session.getValue('profileDescription'),
+      buttonText: this.workflow.session.getValue('buttonText'),
```
Comment (Aierie) [inline packages/web-client/app/components/card-space/create-space-workflow/index.ts:245]: @jurgenwerk I've introduced the workflow session keys 'coverPhotoUrl' and 'profileDescription' here. Realising that our naming of keys for this workflow is not very consistent. Should all keys be prepended with profile?

File: packages/web-client/tests/acceptance/create-card-space-test.ts [hunk 2]
```diff
@@ -169,6 +201,31 @@ module('Acceptance | create card space', function (hooks) {
...
+      .dom('[data-test-sidebar-preview-body] [data-test-profile-card-host]')
+      .containsText('blank.card.space');
+    assert
+      .dom('[data-test-sidebar-preview-body] [data-test-profile-card-category]')
+      .containsText('Category');
+    assert
+      .dom(
+        '[data-test-sidebar-preview-body] [data-test-profile-card-description]'
+      )
+      .containsText('Description');
```
Comment (Aierie) [inline packages/web-client/tests/acceptance/create-card-space-test.ts:222]: Unsure if this particular one is required.

**Comment by backspace** [comment]
I tried this locally, works great! I haven’t looked at the code yet but will now. Thoughts on making the drop shadow match the others in the sidebar?

image

File: packages/web-client/tests/integration/components/workflow-thread-test.ts [hunk 3]
```diff
@@ -290,4 +290,43 @@ module('Integration | Component | workflow-thread', function (hooks) {
...
+
+    await render(hbs`
+      <WorkflowThread @workflow={{this.workflow}} />
+    `);
+
+    assert.dom('[data-test-sidebar-preview-title]').doesNotExist();
+    assert.dom('[data-test-sidebar-preview-body]').doesNotExist();
+    assert.dom('[data-test-sidebar-preview-description]').doesNotExist();
+    assert.dom('[data-test-sidebar-preview-separator]').doesNotExist();
+  });
```
Comment (backspace) [inline packages/web-client/tests/integration/components/workflow-thread-test.ts:331]: Nice to have these tests! 🎉

**Review by backspace — APPROVED** [review]

File: packages/web-client/app/modifiers/scale-down-to-width.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  let difference = originalHeight * (1 - ratio);
+  element.style.marginBottom = `-${difference}px`;
+
+  (child as HTMLElement).style.transform = 'scale(' + ratio + ')';
+  (child as HTMLElement).style.transformOrigin = 'top left';
+}
+
+export default modifier(scaleDownToWidth);
```
[the remark this answers is not in the document]
Comment (jurgenwerk) [reply packages/web-client/app/modifiers/scale-down-to-width.ts:60]: Neat!

File: packages/web-client/app/components/card-space/create-space-workflow/index.ts (hunk 1 above)
Comment (jurgenwerk) [reply packages/web-client/app/components/card-space/create-space-workflow/index.ts:245]: Nice. Yeah would be good for consistency (also to match the naming with the fields from the hub).

**Review by jurgenwerk — APPROVED** [review]
That was quick! Looking good.

**Comment by Aierie** [comment]
> I tried this locally, works great! I haven’t looked at the code yet but will now. Thoughts on making the drop shadow match the others in the sidebar?

This could be good.. though it requires some deep selection to override a child's shadow

**Pull request merged**